hswidget/cofwidget/helpers.go

```diff
@@ -0,0 +1,58 @@
+package cofwidget
+
+import (
+	"github.com/ianling/giu"
+	"github.com/ianling/imgui-go"
+
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2fileformats/d2cof"
+)
+
+//nolint:unparam // width and height are always 15 at the time of writing, but may change
+func makeImageButton(id string, w, h int, t *giu.Texture, fn func()) giu.Layout {
+	return giu.Layout{
+		giu.ImageButton(t).Size(float32(w), float32(h)).OnClick(fn),
+		giu.Custom(func() {
+			// make this button unique across all editor instances
+			// at the time of writing, ImageButton uses the texture ID as the button ID
+			// so it wont be unique across multiple instances if we use the same texture...
+			// we need to step over giu and manually tell imgui to pop the last ID and
+			// push the desired one onto the stack
+			imgui.PopID()
+			imgui.PushID(id)
+		}),
+	}
+}
```

Comment:
maybe we should place it somewhere in hsutil?

Reply:
really, `speedToFPS` and `calculateDuration` need to be a methods of `d2cof.COF` (with better names, of course).

`makeImageButton` needs to be fixed in giu, because we are literally side-stepping giu because we're not allowed to specify the ID of the ImageButton (it uses the texture id as the button id...)

@AllenDeng fyi

Reply:
yah, but `makeImageButton` could be a method of `hscommon` for now, because we'll be able to use it in another packages.

Reply:
I think the ImageButton hack can be moved but the other stuff has no reason to be exported.

Reply:
https://github.com/OpenDiablo2/OpenDiablo2/issues/1058
this is resolved when we add those methods to `d2cof.COF`